### ISSUE TITLE
Predictions extracted with predict_response are not consistent when changing the number of extracts

### ISSUE THREAD

**Issue by VickTaxiarchi** [strengejacke/ggeffects#601]
Hi,

I run a simple model: 

> fit <- svyglm(outc ~ ns(year, df = 4) + cov1 + cov2, family=binomial(link="logit"), design=design_em)

with year ranging from 2010 to 2020. 

My aim is to extract 1000 marginal predictions of the outcome within the range year. Ie marginal prediction of outcome when year = 2010, year = 2010.013, year=2010.26, etc. That is to create a plot of a smoother line of the trend of the outcome. 

To do that:
1. I created a small loop to predict each of these values and save them in a data.frame. So I had 1000 predictions of outcome over time.
> year_seq <- seq(from = 2010, to = 2020, length.out = 1000)

> foreach(value = year_seq) %do% {
  # Generate the prediction for the current year
  prediction <- predict_response(fit , terms = c(paste0("year [", value, "]")))
  
  # Assign the prediction to a dynamically named variable
  assign(paste0("pred", value), prediction, envir = .GlobalEnv)
  
  # Add the current prediction and year to the data frame
  predictions_df <- rbind(predictions_df, data.frame(Year = value, Prediction = prediction))
}

2. To validate the results, I also extracted the predictions of the exact years 2010, 2011, 2012, ... , 2020, as is the original variable in the data.

pred2020<-predict_response(fit, terms=c("year [2010:2020]"))

But the results differed. 

3. Also also extracted up to 2019:

pred2019<-predict_response(fit, terms=c("year [2010:2019]"))

and these again differed to the data frame pred2020.

Therefore my questions are:
(i) why the predictions differ
(ii) how can I now which are the correct ones
(iii) how can I extract the predictions in 1) correctly, since that is my main aim. 

Many thanks
Vicky

**Comment by strengejacke** [comment]
Do you have a reproducible example? It's hard to find out where potential bug are without some more details and a toy example to investigate further.

**Comment by VickTaxiarchi** [comment]
Fair enough. Please see attached test data and my code. I simplified it a bit and instead of length.out=1000 changed it to 20 for faster, but essentially should be the same.

Thank you in advance.

Best wishes
Vicky

Vicky (Paraskevi) Taxiarchi
Research Fellow
Centre for Women’s Mental Health | Division of Psychology and Mental Health | Faculty of Biology, Medicine and Health | University of Manchester |  Manchester | UK

**Comment by VickTaxiarchi** [comment]
No prob. Code in the word file. 
[my_data.csv](https://github.com/user-attachments/files/17679458/my_data.csv)
[Code for GitHub.docx](https://github.com/user-attachments/files/17679486/Code.for.GitHub.docx)

**Comment by strengejacke** [comment]
I see what you would like to achieve, more values for a smoother plot. There must be something wrong in your for-loop, but you can avoid this and add the sequence directly:

``` r
library(survey)
library(splines)
library(ggeffects)

my_data <- datawizard::data_read("~/../Downloads/my_data.csv")

design <- svydesign(ids = ~1, data = my_data, weights = ~weights)

fit <- svyglm(
  outc ~ ns(year, df = 4) + cov1 + cov2 + cov3, 
  family = binomial(), 
  design = design
)
#> Warning in eval(family$initialize): non-integer #successes in a binomial glm!

pred_df <- predict_response(fit, terms = "year [2010:2020 by=0.1]")
plot(pred_df)
```

![](https://i.imgur.com/GOdJVGe.png)

or you create a vector and add that one to the `terms` argument:

``` r
values <- seq(2010, 2020, by = 0.1)
pred_df <- predict_response(fit, terms = "year [values]")
plot(pred_df)
```

![](https://i.imgur.com/7mSDH3m.png)

**Comment by strengejacke** [comment]
To get proper breaks, simply add a ggplot-layer:

``` r
plot(pred_df) + ggplot2::scale_x_continuous(breaks = seq(2010, 2020, 2))
```

![](https://i.imgur.com/F0P8q3j.png)

**Comment by VickTaxiarchi** [comment]
Oh gee thank you. I was trying to find if there was such an option, could see it anywhere. Glad it was so simple.

Many thanks

Best wishes
Vicky

Vicky (Paraskevi) Taxiarchi
Research Fellow
Centre for Women’s Mental Health | Division of Psychology and Mental Health | Faculty of Biology, Medicine and Health | University of Manchester |  Manchester | UK
